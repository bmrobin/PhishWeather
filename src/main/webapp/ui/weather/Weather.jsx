import React from 'react';

export function Weather(props) {
  return (
    <table>
      <tbody>
        <tr>
          <th>Weather for Show <span id="location-id"></span></th>
        </tr>
        <tr>
          <td>Temperature (F)</td>
          <td id="temperature-id">{props.weather.averageTempFahrenheit}</td>
        </tr>
        <tr>
          <td>Date</td>
          <td id="time-id">{props.weather.date}</td>
        </tr>
      </tbody>
    </table>
  );
}
