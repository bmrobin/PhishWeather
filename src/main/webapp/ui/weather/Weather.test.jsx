import React from 'react';
import renderer from 'react-test-renderer';
import { Weather } from './Weather';

describe("Weather component", () => {

  test("should render the weather component", () => {
    let weather = {
      averageTempFahrenheit: 60,
      date: '2017-07-10'
    };
    let component = renderer.create(
      <Weather weather={weather} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

});
