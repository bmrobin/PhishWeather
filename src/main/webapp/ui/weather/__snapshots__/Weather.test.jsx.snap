// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather component should render the weather component 1`] = `
<table>
  <tbody>
    <tr>
      <th>
        Weather for Show 
        <span
          id="location-id"
        />
      </th>
    </tr>
    <tr>
      <td>
        Temperature (F)
      </td>
      <td
        id="temperature-id"
      >
        60
      </td>
    </tr>
    <tr>
      <td>
        Date
      </td>
      <td
        id="time-id"
      >
        2017-07-10
      </td>
    </tr>
  </tbody>
</table>
`;
